Reading List/Reading List/BookController.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  BookController.swift
+//  Reading List
+//
+//  Created by Cora Jacobson on 6/6/20.
+//  Copyright © 2020 Lambda School. All rights reserved.
+//
+
+import Foundation
+
+class BookController {
+    
+    init() {
+        loadFromPersistentStore()
+    }
+    
+    var books: [Book] = []
+    
+    func createBook(_ title: String, _ reasonToRead: String) {
+        let book = Book(title, reasonToRead)
+        books.append(book)
+        saveToPersistentStore()
+    }
+    
+    func deleteBook(_ book: Book) {
+        if let bookIndex = books.firstIndex(of: book) {
+            books.remove(at: bookIndex)
+        }
+        saveToPersistentStore()
+    }
+    
+    func updateHasBeenRead(_ book: Book) {
+        if let bookIndex = books.firstIndex(of: book) {
+            switch books[bookIndex].hasBeenRead {
+            case true:
+                books[bookIndex].hasBeenRead = false
+            case false:
+                books[bookIndex].hasBeenRead = true
+            }
+        }
+        saveToPersistentStore()
+    }
+    
+    func editBook(_ book: Book, _ newTitle: String?, _ newReason: String?) {
+        if let bookIndex = books.firstIndex(of: book) {
+            if let editTitle = newTitle,
+                !editTitle.isEmpty {
+                books[bookIndex].title = editTitle
+            }
+            if let editReason = newReason,
+                !editReason.isEmpty {
+                books[bookIndex].reasonToRead = editReason
+            }
+        }
+        saveToPersistentStore()
+    }
+    
+    var readBooks: [Book] {
+        let readBooksArray = books.filter { $0.hasBeenRead == true }
+        return readBooksArray.sorted(by: { $0.title < $1.title })
+    }
+    
+    var unreadBooks: [Book] {
+        let unreadBooksArray = books.filter { $0.hasBeenRead == false }
+        return unreadBooksArray.sorted(by: { $0.title < $1.title })
+    }
+    
+    // MARK: Persistence
```

Comment:
All of these persistence methods are nicely written. Clean and concise code, with logging of the errors.